app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/RegistrationConnector.scala

```diff
@@ -0,0 +1,98 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.connectors
+
+import com.codahale.metrics.Timer
+import com.kenshoo.play.metrics.Metrics
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.plasticpackagingtax.registration.config.AppConfig
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.RegistrationConnector.{
+  createTimerName,
+  findTimerName,
+  updateTimerName
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.{
+  CreateRegistrationRequest,
+  Registration
+}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RegistrationConnector @Inject() (
+  httpClient: HttpClient,
+  appConfig: AppConfig,
+  metrics: Metrics
+)(implicit ec: ExecutionContext) {
+
+  private val createTimer: Timer = metrics.defaultRegistry.timer(createTimerName)
+  private val updateTimer: Timer = metrics.defaultRegistry.timer(updateTimerName)
+  private val findTimer: Timer   = metrics.defaultRegistry.timer(findTimerName)
+
+  def find(
+    id: String
+  )(implicit hc: HeaderCarrier): Future[Either[ServiceError, Option[Registration]]] = {
+    val timer = findTimer.time()
+    httpClient.GET[Option[Registration]](appConfig.pptRegistrationUrl(id))
+      .andThen { case _ => timer.stop() }
+      .map(resp => Right(resp.map(_.toRegistration)))
+      .recover {
+        case ex: Exception =>
+          Left(
+            DownstreamServiceError(s"Failed to retrieve registration, error: ${ex.getMessage}", ex)
+          )
+      }
+  }
+
+  def create(
+    payload: CreateRegistrationRequest
+  )(implicit hc: HeaderCarrier): Future[Either[ServiceError, Registration]] = {
+    val timer = createTimer.time()
+    httpClient.POST[CreateRegistrationRequest, Registration](appConfig.pptRegistrationUrl, payload)
+      .andThen { case _ => timer.stop() }
+      .map(response => Right(response.toRegistration))
+      .recover {
+        case ex: Exception =>
+          Left(
+            DownstreamServiceError(s"Failed to create registration, error: ${ex.getMessage}", ex)
+          )
+      }
+  }
+
+  def update(
+    payload: Registration
+  )(implicit hc: HeaderCarrier): Future[Either[ServiceError, Registration]] = {
+    val timer = updateTimer.time()
+    httpClient.PUT[Registration, Registration](appConfig.pptRegistrationUrl(payload.id), payload)
+      .andThen { case _ => timer.stop() }
+      .map(response => Right(response.toRegistration))
+      .recover {
+        case ex: Exception =>
+          Left(
+            DownstreamServiceError(s"Failed to update registration, error: ${ex.getMessage}", ex)
+          )
+      }
+  }
+
+}
+
+object RegistrationConnector {
+  val createTimerName = "ppt.create.registration.timer"
+  val updateTimerName = "ppt.update.registration.timer"
+  val findTimerName   = "ppt.find.registration.timer"
```

Comment:
Why do we need public/static variables, via a companion object,  if they are only being used inside methods?
I know they are immutable, but now we have a few code lines handling something that could be handled on the method itself

```
metrics.defaultRegistry.timer(timerName).time()
```

This method `timer(...)` is searching the registry for a timer with a given name if it find returns it, if not, creates one.
There is some overhead of the timer going through a SortedMap, but that is very efficient.

I would move the the timer names into the timer creation themselves of move the all `timer(...)` call into the methods themselves.

If the `timer(...)` call becomes a performance bottleneck. then we could create a reference to the timers.
Otherwise it just seems more boilerplate code to maintain